avocado/utils/pmem.py

```diff
@@ -319,3 +319,109 @@ def destroy_namespace(self, namespace='all', region='', bus='',
                           shell=True, ignore_status=True):
             raise PMemException('Namespace destroy command failed')
         return True
+
+    def write_infoblock(self, namespace='', stdout=False, output=None,
+                        **kwargs):
```

Comment:
Well, while introducing `kwargs` it is a nice solution, this does not reduce the complexity of this method. The "Cognitive Complexity" of this method is still too high (18).

What I was thinking about was to introduce a raw "args" parameter, so we can avoid those if ... else.. inside this method. Something like this:

```python
write_infoblock('foo', args="-p foo -a bar -r foo-bar")
```

If you don't like this idea, another option would be to accept a dictionary with the options:

```python
write_infoblock('foo', args={'-p': 'foo', '-a': 'bar', '-r': 'foo-bar'})
```

In that way, you don't need to write the args inside this method. IMO, this would reduce the complexity here.

What do you think?

Reply:
Although this is a solution to reduce complexity, the idea behind using a method from tests it to make steps simple. IMO, this would mean each tests needs to form something like this `'-p': 'foo', '-a': 'bar', '-r': 'foo-bar'` (dict or string) which kind of makes thing redundant (though possible).

Let us say five different tests uses this function, needs to repeat the parameter with each options, which kind of makes method not of much use except for running a single command.

So is there a better solution which we can adopt here?

Reply:
I see your point and I understand. Lets try to find a better solution. I can see a few spots on this method that could be removed or are odd at least.

So, do you have the man page link for this ndctl that you are using? Because the one that I found has no bus option and do not mention about region and bus being used only with namespace == 'all'.

Reply:
https://pmem.io/ndctl/ndctl-write-infoblock.html
> The keyword all can be specified to operate on every namespace in the system, optionally filtered by bus id (see –bus= option), or region id (see –region= option).

Reply:
@beraldoleal Comments here would be appreciable, Thanks.